src/NUnitFramework/tests/SolutionTests/NuspecDependenciesTests.cs

```diff
@@ -0,0 +1,174 @@
+                {
+                    var condition = itemGroup.Attribute("Condition")?.Value;
+                    string framework = NotSpecified;
+
+                    if (!string.IsNullOrEmpty(condition) && condition.Contains("TargetFrameworkIdentifier"))
+                    {
+                        var split = condition.Split(new[] { "==" }, StringSplitOptions.RemoveEmptyEntries);
+                        framework = split[1].Trim().Replace("'", string.Empty);
+                    }
+
+                    var packageReferences = itemGroup.Descendants("PackageReference")
+                                                     .Select(pr => pr.Attribute("Include")?.Value)
+                                                     .Where(include => !string.IsNullOrEmpty(include)) // Ensure it's non-null and non-empty
+                                                     .Select(include => include!) // Use non-null assertion to ensure the result is a List<string>
+                                                     .ToList();
+
+                    if (!groupedPackages.TryGetValue(framework, out List<string>? referencesForFramework))
+                    {
+                        referencesForFramework = new List<string>();
+                        groupedPackages[framework] = referencesForFramework;
+                    }
+
+                    referencesForFramework.AddRange(packageReferences);
+                }
+
+                return groupedPackages;
+            }
+        }
+
+        internal sealed class NuspecReader
+        {
+            private const string PathToNuspecFolder = $"{Root}/nuget/";
+
+            private string Xml { get; }
+
+            public NuspecReader(string nuspecPath)
+            {
+                var path = Path.GetFullPath(Path.Combine(PathToNuspecFolder, nuspecPath));
+                Assert.That(File.Exists(path), $"Nuspec file at {path} not found.");
+                Xml = File.ReadAllText(path);
+            }
+
+            public Dictionary<string, List<string>> ExtractNuspecPackages()
+            {
+                var doc = XDocument.Parse(Xml);
+                XNamespace ns = "http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd";
+
+                // Dictionary to hold dependencies grouped by targetFramework
+                var groupedDependencies = new Dictionary<string, List<string>>();
+
+                // Find the dependencies section
+                var dependenciesSection = doc.Descendants(ns + "dependencies").FirstOrDefault();
+                if (dependenciesSection is not null)
+                {
+                    // Iterate over all group elements within the dependencies block
+                    foreach (var group in dependenciesSection.Elements(ns + "group"))
+                    {
+                        // Get the targetFramework attribute
+                        var framework = group.Attribute("targetFramework")?.Value ?? "Both";
```

Comment:
What is "Both"? Is this the equivalent of no condition (NotSpecified)? Or is that not allowed in a nuspec?

Reply:
Yes, that should be NotSpecified.

Reply:
But, that should really not happen, or.  That means we would have a section in the nuspec which would apply for all frameworks, if that is even allowed.....